app/repositories/migrations/schema18.ts

```diff
@@ -0,0 +1,458 @@
+    treeType: {type: 'string', default: 'sample'},
+    // stores the additional details for the registration
+    additionalDetails: 'AdditionalDetail[]',
+    // stores the app metadata. Needs to be stringified as it might contain nested array/objects
+    appMetadata: 'string?',
+    // stores the hid when registration is uploaded successfully
+    hid: 'string?',
+    // stores the plant location version history
+    plantLocationHistory: 'PlantLocationHistory[]',
+    //store remeasurement related dates
+    remeasurementDates: 'RemeasurementDates?',
+  },
+};
+
+const Inventory = {
+  name: 'Inventory',
+  primaryKey: 'inventory_id',
+  properties: {
+    inventory_id: 'string',
+    plantation_date: 'date?',
+    treeType: 'string?',
+    status: 'string?',
+    projectId: 'string?',
+    donationType: 'string?',
+    locateTree: 'string?',
+    lastScreen: 'string?',
+    species: 'Species[]',
+    polygons: 'Polygons[]',
+    specieDiameter: 'double?',
+    specieHeight: 'double?', // <*IMPORTANT*> ONLY FOR SINGLE TREE
+    tagId: 'string?',
+    registrationDate: 'date?',
+    // stores the count of sample trees which are to be recorded
+    sampleTreesCount: 'int?',
+    // stores the sample trees having length equal to tree count
+    sampleTrees: 'SampleTrees[]',
+    // stores the number of sample trees which are already recorded
+    completedSampleTreesCount: {
+      type: 'int?',
+      default: 0,
+    },
+    // stores the number of sample trees which are uploaded to server
+    uploadedSampleTreesCount: {
+      type: 'int?',
+      default: 0,
+    },
+    // stores the location id of the plant location which is available
+    // when the inventory data is uploaded
+    locationId: 'string?',
+    // stores the additional details for the registration
+    additionalDetails: 'AdditionalDetail[]',
+    // stores the app metadata. Needs to be stringified as it might contain nested array/objects
+    appMetadata: 'string?',
+    // stores the hid when registration is uploaded successfully
+    hid: 'string?',
+    // stores the original geoJSON of coordinates in string which was uploaded
+    // for the first time for a registration
+    originalGeometry: 'string?',
+    // stores the plant location version history
+    plantLocationHistory: 'PlantLocationHistory[]',
+  },
+};
+
+const User = {
+  name: 'User',
+  primaryKey: 'id',
+  properties: {
+    id: 'string?',
+    accessToken: 'string?',
+    idToken: 'string?',
+    email: 'string?',
+    firstName: 'string?',
+    lastName: 'string?',
+    image: 'string?',
+    country: 'string?',
+    isLogEnabled: 'bool?',
+    userId: 'string?',
+    userType: 'string?',
+    refreshToken: 'string?',
+    isSignUpRequired: 'bool?',
+    type: 'string?',
+    displayName: 'string?',
+    // stores the expiry time of token in seconds
+    expirationTime: 'int?',
+    fetchNecessaryInventoryFlag: {type: 'int', default: InventoryType.NecessaryItems},
```

Comment:
Got a migration warning from Realm for this field. See other error. This error vanished after deleting my old app and starting with a new one having deleted all my local data. I hope Realm database migrations from an older version of the app will work.